io/io/src/TBufferFile.cxx

```diff
@@ -2275,7 +2374,7 @@ Int_t TBufferFile::WriteFastArray(void **start, const TClass *cl, Int_t n,
       (*streamer)(*this,(void*)start,0);
       return 0;
    }
-
+   if (n < 0) return -1;
```

Comment:
Why can't we guard against the 1GB limit in this method too? Also, why are we not throwing `Fatal` here?

Reply:
Same comment as above. It will abort in deeper calls. We decided to throw Fatal as there was no return type in the other functions, compared to this one. Of course, I can throw Fatal here, too, if you prefer.